### PR TITLE
Refactor and extract generateDaySlots function in Availability components

### PR DESCRIPTION
**Pull request by yifanplanet**
# Code changes

- Refactor generateDaySlots function in Availability components and extract it for import
- Extract overlap and arrayContainsArray functions

# Readiness checklist

- [ ] Cypress tests passing?

# License

I confirm that this contribution is made under the terms of the MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: components/availability/src/Availability.svelte [hunk 1]
```diff
@@ -638,7 +444,10 @@
           numFreeCalendars === 0 ||
           fewerCalendarsThanRatio ||
           (_this.required_participants.length &&
-            !allRequiredParticipantsIncluded(epoch[0].available_calendars))
+            !arrayContainsArray(
```
Comment (pooja169usp) [inline components/availability/src/Availability.svelte:447]: `arrayContainsArray` is not clear as to what is being done here. I'd suggest adding a comment.

File: components/availability/src/method/availabilityUtils.ts [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+      ) {
+        availability = AvailabilityStatus.PARTIAL;
+      }
+
+      // If availability is partial, but a required participant is unavailble, the slot becomes Busy
+      if (
+        availability === AvailabilityStatus.PARTIAL &&
+        internalProps.required_participants.length
+      ) {
+        if (!arrayContainsArray(requiredParticipants, freeCalendars)) {
```
Comment (pooja169usp) [inline components/availability/src/method/availabilityUtils.ts:149]: Same comment here

File: components/availability/src/method/availabilityUtils.ts [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+  Calendar,
+  Manifest,
+  TimeSlot,
+  SelectableSlot,
+} from "@commons/types/Availability";
+import { arrayContainsArray } from "@commons/methods/component";
+
+// map over the ticks() of the time scale between your start day and end day
+// populate them with as many slots as your start_hour, end_hour, and slot_size dictate
+export const generateDaySlots = (
```
Comment (krisgardiner) [inline components/availability/src/method/availabilityUtils.ts:18]: Might be worth it to start writing some unit tests for this in `tests`

**Review by krisgardiner — APPROVED** [review]
Thanks for adding tests!

**Review by philrenaud — APPROVED** [review]
Looks good, @yifanplanet !

**Review by pooja169usp — APPROVED** [review]

**Comment by AaronDDM** [comment]
🎉 congrats, first PR merged!

**Pull request merged**